packages/teams-js/src/public/pages.ts

```diff
@@ -177,9 +191,14 @@ export namespace pages {
    * On mobile hosts or hosts where there is no keyboard interaction or UI notion of "focus" this function has no
    * effect and will be a no-op when called.
    * @param navigateForward - Determines the direction to focus in host.
+   * @param returnFocusType - Determines the type of focus to return to in the host.
    */
-  export function returnFocus(navigateForward?: boolean): void {
-    returnFocusHelper(getApiVersionTag(pagesTelemetryVersionNumber, ApiName.Pages_ReturnFocus), navigateForward);
+  export function returnFocus(navigateForward?: boolean, returnFocusType?: ReturnFocusType): void {
```

Comment:
I spent some time thinking about this, and I think you'll need to modify this in order for it to work (although I don't think the modification will be too tricky). There are two sort of competing "tricky parts" of what you're trying to do here:

1) Because there are two optional parameters in this function and `returnFocusType` is meant to supersede `navigateForward`, the current signature makes it pretty easy for developers to unintentionally pass in invalid/illegal combinations of parameters, for example `returnFocus(false, ReturnFocusType.NextLandmark)` is sort of giving opposite directions to the function.
2) This function can't ever *only* send an enum value because there will be some hosts (older versions of Teams for example) that don't know about the enum and *only* know how to handle the boolean.

Given those restrictions, I think the best path forward would be to make function overloads for returnFocus that take in different parameters, and then figure out how to resolve the different signatures and send up only valid combinations of booleans and enums. I spent a few minutes this morning puttering around and came up with this:

```typescript
// Document this function for developers
export function returnFocus(navigateForward?: boolean): void;

// Document this function for developers
// You could choose to make this a required parameter if you want since it's a new signature, but if you
// would rather keep it as optional and then determine what the default should be that's okay with me
export function returnFocus(returnFocusType?: ReturnFocusType): void;

function returnFocus(arg1?: boolean | ReturnFocusType): void {
  if (arg1 === undefined) {
    // send default boolean and default ReturnFocusType values up to host
    // You will need to document clearly what will happen on the host in this case
    // (for example, if neither is specified will *either* go to NextLandmark if
    // it can, and if not will go forward)
    console.log('Sent default boolean value of true and default ReturnFocusType of NextLandmark')
    return;
  }

  if (typeof arg1 === 'boolean') {
    console.log(`Sent boolean ${arg1} and no ReturnFocusType`);
    return;
  } else {
    let backupBooleanValue: boolean;
    switch (arg1) {
      case (ReturnFocusType.PreviousLandmark):
      case (ReturnFocusType.GoToActivityFeed):
        backupBooleanValue = false;
        break;
      case (ReturnFocusType.NextLandmark):
        backupBooleanValue = true;
    }
    console.log(`Sent ReturnFocusType ${arg1} and backup boolean value ${backupBooleanValue}`);
  }
}
```

Essentially, you would have two exported functions. 1 only accepts booleans and 1 only accepts enums. You then have a third function that is *not* exported where all of the actual implementation is stored. This function figures out whether it has been passed a boolean or enum and then passes the *correct* combination of boolean and enum up to the host. There are 3 cases you would need to handle:

1) No parameter is passed in. In this case you will need to send up logical default values. You could either send up *just* a boolean, or you could send a default boolean and a default enum.

2) A boolean is passed in. You probably just send that boolean to the host.

3) An enum is passed in. In order for this to work on older hosts that don't know about the enum, you will need to map each enum value to it's most logical boolean value so that developers who call that function on older hosts get "basically correct" behavior.

I realize this is a lot to read, but I think the change itself is actually not too bad. If you have questions about any of this, just send me a message and we can chat.

Reply:
HI @TrevorJoelHarris, thanks for taking time and review the PR. I tried to incorporate the changes that were called out. Feel free to take another look